src/optimal_retention.rs

```diff
@@ -216,12 +216,15 @@ pub fn simulate(
 
     let mut card_priorities = PriorityQueue::new();
 
-    fn card_priority(card: &Card, learn: bool) -> (usize, bool, usize) {
-        // High difficulty priority as example
-        (-card.due as usize, !learn, -card.difficulty as usize)
+    fn card_priority(card: &Card, learn: bool) -> (i32, bool, i32) {
+        // high priority for early due, review, low difficulty card
+        (-card.due as i32, !learn, -(card.difficulty * 100.0) as i32)
     }
 
     for (i, card) in cards.iter().enumerate() {
+        if card.due >= learn_span as f32 {
+            continue;
+        }
```

Comment:
```suggestion
```

![image](https://github.com/user-attachments/assets/39115917-66e9-4fdd-86e1-d2a2fe5142b5)

Got it.